packages/plans-grid-next/src/hooks/data-store/use-grid-plans.tsx

```diff
@@ -44,36 +47,85 @@ interface Props {
 	// allFeaturesList temporary until feature definitions are ported to calypso-products package
 	allFeaturesList: FeatureList;
 	useCheckPlanAvailabilityForPurchase: Plans.UseCheckPlanAvailabilityForPurchase;
-	useFreeTrialPlanSlugs: UseFreeTrialPlanSlugs;
-	eligibleForFreeHostingTrial: boolean;
+	useFreeTrialPlanSlugs?: UseFreeTrialPlanSlugs;
+	eligibleForFreeHostingTrial?: boolean;
 	storageAddOns: ( AddOnMeta | null )[] | null;
 	selectedFeature?: string | null;
 	term?: ( typeof TERMS_LIST )[ number ]; // defaults to monthly
 	intent?: PlansIntent;
 	selectedPlan?: PlanSlug;
-	sitePlanSlug?: PlanSlug | null;
-	hideEnterprisePlan?: boolean;
+	hiddenPlans?: HiddenPlans;
 	isInSignup?: boolean;
 	showLegacyStorageFeature?: boolean;
-
+	isDisplayingPlansNeededForFeature?: boolean;
 	/**
 	 * If the subdomain generation is unsuccessful we do not show the free plan
 	 */
 	isSubdomainNotGenerated?: boolean;
 	coupon?: string;
-	selectedSiteId?: number | null;
+	siteId?: number | null;
 }
 
+const isGridPlanVisible = ( {
+	hiddenPlans: {
+		hideFreePlan,
+		hidePersonalPlan,
+		hidePremiumPlan,
+		hideBusinessPlan,
+		hideEcommercePlan,
+	} = {},
+	isDisplayingPlansNeededForFeature,
+	planSlug,
+	planSlugsForIntent,
+	selectedPlan,
+}: {
+	hiddenPlans?: HiddenPlans;
+	isDisplayingPlansNeededForFeature?: boolean;
+	planSlug: PlanSlug;
+	planSlugsForIntent: PlanSlug[];
+	selectedPlan?: PlanSlug;
+} ): boolean => {
```

Comment:
Maybe we can attempt a different refactor here. If we pass a `usePlanVisibilityCheck` (or something of this nature), that returns a callback, then it looks like we'd be able to avoid passing either `isDisplayingPlansNeededForFeature` or `hiddenPlans` into the grid components.

Reply:
>  avoid passing either isDisplayingPlansNeededForFeature or hiddenPlans into the grid components

Just checking that what you meant here are the grid 'hooks' rather than the components, as the components don't currently use these props?

I've started on this refactor ([361a4b0](https://github.com/Automattic/wp-calypso/pull/87478/commits/361a4b0a2cd5277c3650a3c2c6a0b5f503235b96)), is that the general direction you had in mind? Working through this I was reminded that `hiddenPlans` is also used by the hook `usePlanTypesWithIntent`, so I've introduced a new `isEnterprisePlanAvailable` param. That doesn't seem ideal, though, so I'll revisit with fresh eyes next week.

Reply:
Ok. I feel we may have been approaching this from a slightly wrong/unintended angle. I've looked this over and needed to get my thinking back in place a bit. Apologies if I've been misguiding at all.

My original intention with the comment in the code was to try and deprecate any of the unused "hidePersonal", "hideBusiness", etc. flags, and for whatever remained to become part of `useGridPlans`. This would mean getting rid of `isVisible` props from the `GridPlan` object altogether; cause this is where it all leads to - we first grab the plans for the intent, then filter via `useFilterPlansForPlanFeatures` (which maps them over these various flags), and then remove them from the list in the individualized hooks (`useGridPlansForComparisonGrid`, etc.). It's this flow that I wanted to narrow into and see if possible to improve. 

I think these questions are worth asking:

- which `hidden*` are effectively used and cannot be deprecated? (I think there is only one or two that cannot be mapped onto a "plans-intent")
- is `isVisible` property on the grid-plans used anywhere else i.e. does a plan with `isVisible: false` have a place in the UI?

What we've been doing with this refactor so far is migrating the entirety of `useFilterPlansForPlanFeatures` into the main hook, which isn't quite serving much. Not is this update above IMO, as it's also a bit of indirection.

I can take another look to confirm, but I think for now, and to get this moving forward, we can go back to either passing `hiddenPlans` into the hooks or roll back to when `useFilterPlansForPlanFeatures` was living in `plans-features-main`. I would maybe go for the latter. Let's also add back the comment in the code if we are revisiting later.

Reply:
Ok I've rolled back this commit/refactor: [361a4b0](https://github.com/Automattic/wp-calypso/pull/87478/commits/361a4b0a2cd5277c3650a3c2c6a0b5f503235b96)

> What we've been doing with this refactor so far is migrating the entirety of useFilterPlansForPlanFeatures into the main hook, which isn't quite serving much - but it does centralize things and make plans-features-main leaner. 🤔

Yes in my mind making `plans-features-main` leaner has been a key goal, as it represents clear progress towards unlocking the reusability of the grids. 

I think the 'visibility' refactoring you've described here deserves a separate issue. I've created one using your comments above, but please update/change as needed as you have the clearest picture of what needs to be done: https://github.com/Automattic/wp-calypso/issues/88403